tools/build_defs/repo/git.bzl

```diff
@@ -37,18 +39,33 @@ def _clone_or_update_repo(ctx):
         fail("Exactly one of commit, tag, or branch must be provided")
 
     root = ctx.path(".")
-    directory = str(root)
-    if ctx.attr.strip_prefix:
-        directory = root.get_child(".tmp_git_root")
-
-    git_ = git_repo(ctx, directory)
+    git_ = git_repo(ctx, root)
 
     if ctx.attr.strip_prefix:
-        dest_link = "{}/{}".format(directory, ctx.attr.strip_prefix)
-        if not ctx.path(dest_link).exists:
+        strip_prefix_dir = root.get_child(ctx.attr.strip_prefix)
+        if not strip_prefix_dir.exists:
             fail("strip_prefix at {} does not exist in repo".format(ctx.attr.strip_prefix))
-        for item in ctx.path(dest_link).readdir():
-            ctx.symlink(item, root.get_child(item.basename))
+        if not strip_prefix_dir.is_dir:
+            fail("strip_prefix at {} is not a directory".format(ctx.attr.strip_prefix))
+
+        # 1. Move the contents of the strip_prefix subdirectory to a hidden
+        #    directory in the root.
+        ctx.rename(strip_prefix_dir, root.get_child(_STRIP_PREFIX_TMP_DIR))
+
+        # 2. Delete all other top-level entries.
+        for entry in root.readdir():
+            if entry.basename != _STRIP_PREFIX_TMP_DIR and entry.basename != ".git":
```

Comment:
why leave the `.git` directory behind? If we need to call git commands, we should probably do it before all the strip_prefix stuff is performed. (Note that in the old version, the `.git` directory is inside `.tmp_git_root` so git commands could be called in there safely.)